public/app/plugins/panel/timeseries/plugins/AnnotationsPlugin.tsx

```diff
@@ -28,9 +29,16 @@ export const AnnotationsPlugin = ({ annotations, timeZone, config }: Annotations
     }
 
     annotationsRef.current = views;
+
+    return () => {
+      // clear on unmount
+      annotationsRef.current = [];
+    };
   }, [annotations]);
 
   useLayoutEffect(() => {
+    getTimeSrv().refreshTimeModel();
```

Comment:
I tried testing this without this line and it seems to work as expected 🤔 Is this change necessary?


https://user-images.githubusercontent.com/22381771/236925439-7756e8a5-2ab0-4fc8-b22e-e3c0d0b376d5.mov

Reply:
This auto-refreshes the dashboard after adding/deleting a new annotation - otherwise you'd have to manually refresh from the button. Not sure if there were any complaints or if it's a requirement, seeing @ryantxu comment might be a good idea to address it separately..

Reply:
yes -- that should be addressed at the dashboard level, not within the panel